test/e2e/functional/tests/e2e.go

```diff
@@ -354,6 +365,826 @@ var _ = Describe("Ingress Node Firewall", func() {
+				}
+				return true
+			}, timeout, retryInterval)
+		})
+
+		It("update to block target port in a port range", func() {
+			Eventually(func() bool {
+				inf, err = infwutils.GetIngressNodeFirewall(testclient.Client, inf.Name, inf.Namespace, timeout)
+				if err != nil {
+					return false
+				}
+				infwutils.RedefineIngressTCPRulePort(inf, 0, 0, "79-81")
+				err = infwutils.UpdateIngressNodeFirewall(testclient.Client, inf, timeout)
+				if err != nil {
+					return false
+				}
+				return true
+			}, timeout, retryInterval)
+
+			Eventually(func() bool {
+				_, stdErr, err := pods.NetcatTCPV4(testclient.Client, client, server.Status.PodIP, defaultSourcePort,
+					serverListenPort, timeout)
+				if stdErr != "" || err != nil {
+					return true
+				}
+				return false
+			}, timeout, retryInterval)
+		})
+
+		It("update to block port in non-first order rule", func() {
+			Eventually(func() bool {
+				inf, err = infwutils.GetIngressNodeFirewall(testclient.Client, inf.Name, inf.Namespace, timeout)
+				if err != nil {
+					return false
+				}
+				// no longer block in first rule
+				infwutils.RedefineIngressTCPRulePort(inf, 0, 0, "79")
+				//block in second rule
+				infwutils.AppendIngressRule(inf, 0, infwutils.GetTCPRule(2, serverListenPort,
+					ingressnodefwv1alpha1.IngressNodeFirewallDeny))
+				err = infwutils.UpdateIngressNodeFirewall(testclient.Client, inf, timeout)
+				if err != nil {
+					return false
+				}
+				return true
+			}, timeout, retryInterval)
+
+			Eventually(func() bool {
+				_, stdErr, err := pods.NetcatTCPV4(testclient.Client, client, server.Status.PodIP, defaultSourcePort,
+					serverListenPort, timeout)
+				if stdErr != "" || err != nil {
+					return true
+				}
+				return false
+			}, timeout, retryInterval)
+		})
+
+		It("block new target with update to sourceCIDRs", func() {
+			additionalServer := pods.GetNetcatServerPod(additionalServerPodName, OperatorNameSpace, serverListenPort, podLabel,
+				ingressnodefwv1alpha1.ProtocolTypeTCP)
+			additionalServer, err = pods.EnsurePodRunning(testclient.Client, additionalServer, OperatorNameSpace, retryInterval, timeout)
+			defer pods.DeletePod(testclient.Client, additionalServer, timeout)
+			Expect(err).Should(BeNil())
+			Eventually(func() bool {
+				inf, err = infwutils.GetIngressNodeFirewall(testclient.Client, inf.Name, inf.Namespace, timeout)
+				if err != nil {
+					return false
+				}
+				infwutils.AppendFirstIngressSourceCIDRs(inf, additionalServer.Status.PodIP)
+				err = infwutils.UpdateIngressNodeFirewall(testclient.Client, inf, timeout)
+				if err != nil {
+					return false
+				}
+				return true
+			}, timeout, retryInterval)
+
+			Eventually(func() bool {
+				_, stdErr, err := pods.NetcatTCPV4(testclient.Client, additionalServer, server.Status.PodIP, defaultSourcePort,
+					serverListenPort, timeout)
+				if stdErr != "" || err != nil {
+					return true
+				}
+				return false
+			}, timeout, retryInterval)
+		})
```

Comment:
if strongly feel we can do this as table we pass in the rule and the expected result as `table.Entry` that will cut down on code duplication and make it easy to extend new cases for different protocols ?